### PR TITLE
tests: addition of test_example_app.py

### LINKED ISSUE #108: tests: addition of test_example_app.py
The package contains `examples/app.py` example application, but there is no test
for it in `tests/test_example_app.py`.  It should be added.  See existing examples:

- https://github.com/inveniosoftware/invenio-records/blob/master/tests/test_example_app.py
- https://github.com/inveniosoftware/invenio-marc21/blob/master/tests/test_example_app.py

### PR DESCRIPTION
**Pull request by hachreak**
* Adds test_example_app.py. (closes #108)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

**Comment by hachreak** [comment]
ping @jirikuncar

**Comment by jirikuncar** [comment]
@hachreak you can rebase now. Thanks!

**Comment by hachreak** [comment]
ping @jirikuncar

File: examples/app-recreate.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
```
Comment (jirikuncar) [inline examples/app-recreate.sh]: What about calling the file `app-setup.sh` and having also `app-teardown.sh` with `flask db destroy --yes-i-know`?

File: examples/app-setup.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+export FLASK_APP=app.py
+
+# Install specific dependencies
+pip install -r requirements.txt || exit 1
```
Comment (jirikuncar) [inline examples/app-setup.sh]: What about using [`set -e`](http://www.gnu.org/software/bash/manual/bashref.html#The-Set-Builtin) instead of `|| exit 1`?

**Comment by hachreak** [comment]
@jirikuncar could you please have a look if it's ok.. them I'll change all PR for example apps as I did here

File: examples/app-setup.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+set -e
```
Comment (jirikuncar) [inline examples/app-setup.sh]: 👍

File: examples/app.py [hunk 4]
```diff
@@ -27,18 +27,23 @@
...
+    $ pip install -e .[all]
     $ cd examples
-    $ pip install -r requirements.txt
-    $ export FLASK_APP=app.py
-    $ flask db init
-    $ flask db create
-    $ flask index init
-    $ flask fixtures records
-    $ flask run --debugger
+    $ # clean the environment if needed
```
Comment (jirikuncar) [inline examples/app.py]: I would just move it at the end of the docstring so people don't loose the focus.

Comment (hachreak) [inline examples/app.py]: As a ".. note::"?

Comment (jirikuncar) [inline examples/app.py]: It's not really important since we are not including it in Sphinx.

File: examples/app-setup.sh (hunk 3 above)
Comment (tiborsimko) [inline examples/app-setup.sh]: If we standardise on shell scripts for testing example app, then it's good to add also an unbound symbol check, to be future-proof.  See e.g. [Invenio scripts](https://github.com/inveniosoftware/invenio/blob/master/scripts/populate-instance.sh#L98-L100) where I'm using:

```
# quit on errors:
set -o errexit

# quit on unbound symbols:
set -o nounset
```

But let me reply in the other shell-scripting-vs-Python-scripting thread...

File: .travis.yml [hunk 5]
```diff
@@ -46,8 +47,6 @@ env:
 
 python:
   - "2.7"
-  - "3.3"
-  - "3.4"
```
Comment (tiborsimko) [inline .travis.yml]: Let's remove 3.3 and 3.4 also from `CONTRIBUTING.rst`.

File: examples/app-setup.sh (hunk 3 above)
Comment (jirikuncar) [inline examples/app-setup.sh]: I was proposing this as an alternative to `|| exit 1` which was used before ...

File: .travis.yml (hunk 5 above)
Comment (hachreak) [inline .travis.yml]: done

**Pull request merged**